co_sim_io/impl/communication/file_communication.hpp

```diff
@@ -352,6 +352,7 @@ class FileCommunication : public Communication
             }
         }
 
+        input_file.close();
```

Comment:
this was the problem @pooyan-dadvand 

Guess we were lucky that it worked by coincidence in the other cases
=> good to know also for the future